flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/CliClient.java

```diff
@@ -414,25 +418,33 @@ private void callSelect(QueryOperation operation) {
     private boolean callInsert(CatalogSinkModifyOperation operation) {
         printInfo(CliStrings.MESSAGE_SUBMITTING_STATEMENT);
 
+        TableResult tableResult = null;
+        boolean sync = executor.getSessionConfig(sessionId).get(TABLE_DML_SYNC);
+        String message;
+        if (sync) {
+            printInfo(MESSAGE_WAIT_EXECUTE);
+            message = MESSAGE_FINISH_STATEMENT;
+        } else {
+            message = MESSAGE_STATEMENT_SUBMITTED;
+        }
         try {
-            TableResult result = executor.executeOperation(sessionId, operation);
-            checkState(result.getJobClient().isPresent());
-            terminal.writer()
-                    .println(
-                            CliStrings.messageInfo(CliStrings.MESSAGE_STATEMENT_SUBMITTED)
-                                    .toAnsi());
+            tableResult = executor.executeOperation(sessionId, operation);
+            checkState(tableResult.getJobClient().isPresent());
+            terminal.writer().println(CliStrings.messageInfo(message).toAnsi());
+
             // keep compatibility with before
             terminal.writer()
                     .println(
                             String.format(
                                     "Job ID: %s\n",
-                                    result.getJobClient().get().getJobID().toString()));
+                                    tableResult.getJobClient().get().getJobID().toString()));
```

Comment:
I think we don't need to print the job id at sync mode, the job id looks redundant because job has been finished.